# CrunchieMunchies

You work in marketing for a food company <b>myCorps</b>, which is developing a new kind of tasty, wholesome cereal called <b>CrunchieMunchies</b>. 

You want to demonstrate to consumers how healthy your cereal is in comparison to other leading brands, so you’ve dug up nutritional data on several different competitors.

Your task is to use <em>NumPy statistical calculations</em> to analyze this data and prove that your <b>CrunchieMunchies</b> is the healthiest choice for consumers.






# Task STEPS


1.First, import numpy.

In [8]:
import numpy as np



2.Look over the <b><em>cereal.csv</em></b> file. This file contains the reported calorie amounts for different cereal brands. Load the data from the file and save it as <b><em>calorie_stats.</em></b>



In [9]:
calorie_stats = np.genfromtxt('cereal.csv', delimiter=",")
calorie_stats



array([ 70., 120.,  70.,  50., 110., 110., 110., 130.,  90.,  90., 120.,
       110., 120., 110., 110., 110., 100., 110., 110., 110., 100., 110.,
       100., 100., 110., 110., 100., 120., 120., 110., 100., 110., 100.,
       110., 120., 120., 110., 110., 110., 140., 110., 100., 110., 100.,
       150., 150., 160., 100., 120., 140.,  90., 130., 120., 100.,  50.,
        50., 100., 100., 120., 100.,  90., 110., 110.,  80.,  90.,  90.,
       110., 110.,  90., 110., 140., 100., 110., 110., 100., 100., 110.])

3.There are <em>60 calories per serving of CrunchieMunchies</em>. How much <b>higher</b> is the <b>average calorie count</b> of your competition?

Save the answer to the variable <b>average_calories</b> and print the variable to the terminal to see the answer.

In [10]:
average_calories = np.mean(calorie_stats) - 60
print(average_calories)


46.883116883116884


4.Does the <b>average calorie count</b> adequately reflect the distribution of the dataset? Let’s sort the data and see.

<b><em>Sort</em></b> the data and save the result to the variable <b>calorie_stats_sorted</b>. Print the sorted data to the terminal.


In [11]:
calorie_stats_sorted = np.sort(calorie_stats)

print(calorie_stats_sorted)

[ 50.  50.  50.  70.  70.  80.  90.  90.  90.  90.  90.  90.  90. 100.
 100. 100. 100. 100. 100. 100. 100. 100. 100. 100. 100. 100. 100. 100.
 100. 100. 110. 110. 110. 110. 110. 110. 110. 110. 110. 110. 110. 110.
 110. 110. 110. 110. 110. 110. 110. 110. 110. 110. 110. 110. 110. 110.
 110. 110. 110. 120. 120. 120. 120. 120. 120. 120. 120. 120. 120. 130.
 130. 140. 140. 140. 150. 150. 160.]


5.Do you see what I’m seeing? Looks like <b><em>the majority of the cereals are higher than the mean</em></b>. Let’s see if the <b>median</b> is a better representative of the dataset.

Calculate the median of the dataset and save your answer to <b><em >median_calories</em></b>. Print the median so you can see how it compares to the mean.

In [12]:
median_calories = np.median(calorie_stats)
print('median:', median_calories,'-- mean:',np.mean(calorie_stats))


median: 110.0 -- mean: 106.88311688311688


6.While the median demonstrates that <b><em><q>at least half of our values are over 100 calories</q></em></b>, it would be more impressive to show that a significant portion of the competition has a higher calorie count that CrunchieMunchies.

<b>Calculate different percentiles</b> and print them to the terminal until you find the lowest percentile that is greater than 60 calories. Save this value to the variable <b>nth_percentile</b>.


In [28]:
a = np.arange(1,101)
c = np.percentile(calorie_stats_sorted,a) 
print(c)
#calorie_stats_sorted [calorie_stats_sorted]
nth_percentile = c[c>60].min()

print("Lowest nth_percentile greater than 60 is: ", nth_percentile,  ", at nth number: ", a[c>60].min())


[ 50.   50.   55.6  70.   70.   75.6  83.2  90.   90.   90.   90.   90.
  90.   90.   90.   91.6  99.2 100.  100.  100.  100.  100.  100.  100.
 100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.
 100.  100.  106.4 110.  110.  110.  110.  110.  110.  110.  110.  110.
 110.  110.  110.  110.  110.  110.  110.  110.  110.  110.  110.  110.
 110.  110.  110.  110.  110.  110.  110.  110.  110.  110.  110.  110.
 110.  110.  110.  110.  115.2 120.  120.  120.  120.  120.  120.  120.
 120.  120.  120.  120.  120.  124.  130.  130.  136.8 140.  140.  140.
 147.2 150.  152.4 160. ]
Lowest nth_percentile greater than 60 is:  70.0 , at nth number:  4


7.While the percentile shows us that<b><em><q>the majority of the competition has a much higher calorie count</q></em></b>, it’s an awkward concept to use in marketing materials.

Instead, let’s calculate the percentage of cereals that <b><em><q>have more than 60 calories per serving</q></em></b>. Save your answer to the variable <b><em>more_calories</em></b> and print it to the terminal

In [26]:
more_calories = np.mean(calorie_stats > 60)
more_calories


0.961038961038961

8.Wow! That’s a really high percentage. That’s going to be very useful when we promote CrunchieMunchies. But one question is, <b><em>how much variation exists in the dataset? </b></em></q>Can we make the generalization that most cereals have around 100 calories or is the spread even greater?

Calculate the amount of variation by finding the <b><em>standard deviation</em</b> Save your answer to <b><em>calorie_std</em></b> and print to the terminal. How can we incorporate this value into our analysis?

In [26]:
calorie_std = np.std(calorie_stats_sorted)
calorie_std


19.35718533390827

9.Write a short paragraph that sums up your findings and how you think this data could be used to 
<b>myCorp’s</b> advantage when marketing CrunchieMunchies.


The findings show calories comparison of <b> CrunchieMunchies </b> and its competitors' cereals. <b> CrunchieMunchies </b> carries less calories than 96% of other cereals. For other cereals, the median calorie count centres at 110 and the mean at 106.88 having a standard deviation of 19.35. In contrast with its competitors' cereals, <b> CrunchieMunchies' </b> calorie count places it at 4th percentile leaving only 3 competiotrs having lesser calories. It seems <b> CrunchieMunchies </b> will fare well in the market and it'll attract huge attention of health conscious consumers. It has a great potential for success.
 